### PR TITLE
MPU initialization refactoring

### PR DESCRIPTION
**Pull request by cepetr**
I've refactored MPU initialization routine to be more clear and maintainable. Now it's just a draft for further discussion.

### CODE REVIEW

**Review by TychoVrahe — COMMENTED** [review]
Couple clarifications for the problems discovered during refactoring below.

Note that the region sizes will need adjustments for final MCU variants (U585 and U5Gx likely).

File: core/embed/trezorhal/stm32u5/mpu.c [hunk 1]
```diff
@@ -18,241 +18,163 @@
...
+  //   REGION    ADDRESS                   SIZE                TYPE       WRITE   UNPRIV 
+  SET_REGION( 0, FLASH_BASE_S,             SIZE_16K,           FLASH_DATA,  YES,   YES ); // Secret 
+  SET_REGION( 1, FLASH_BASE_S + SIZE_16K,  SIZE_48K,           FLASH_CODE,   NO,   YES ); // Boardloader code 
+  SET_REGION( 2, FLASH_BASE_S + SIZE_64K,  SIZE_4032K,         FLASH_DATA,  YES,   YES ); // Bootloader + Storage + Firmware 
+  SET_REGION( 3, SRAM1_BASE_S,             SIZE_768K,          SRAM,        YES,   YES ); // SRAM1 
+  SET_REGION( 4, SRAM2_BASE_S + 0x100,     SIZE_1728K - 0x100, SRAM,        YES,   YES ); // SRAM2/3/5 + stack guard 
+  SET_REGION( 5, GFXMMU_BUFFERS_S,         SIZE_16M,           SRAM,        YES,   YES ); // Frame buffer 
+  SET_REGION( 6, PERIPH_BASE_NS,           SIZE_512M,          PERIPHERAL,  YES,   YES ); // Peripherals NS + S ?? proc NS 
+  SET_REGION( 7, FLASH_OTP_BASE,           FLASH_OTP_SIZE,     FLASH_DATA,  YES,   YES ); // OTP
+//SET_REGION( 7, FLASH_BASE_NS,            SIZE_4M,            FLASH_DATA,  YES,   YES ); // !!!!! prepis regionu 7
```
[the remark this answers is not in the document]
Comment (cepetr) [reply core/embed/trezorhal/stm32u5/mpu.c]: fixed, see 853b1c34b1de7598c26b3671bbade6b27b0984af

File: core/embed/trezorhal/stm32u5/mpu.c [hunk 2]
```diff
@@ -18,241 +18,163 @@
...
-              LL_MPU_INSTRUCTION_ACCESS_DISABLE | SHAREABILITY_FLASH;
-  MPU->RLAR = REGION_END(FLASH_OTP_BASE + FLASH_OTP_SIZE - 1) |
-              ATTR_IDX_FLASH_NON_CACHABLE;
-
-  // Enable MPU
+  mpu_set_attributes();
+  // clang-format off
+  //   REGION    ADDRESS                   SIZE                TYPE       WRITE   UNPRIV 
+  SET_REGION( 0, FLASH_BASE_S,             SIZE_64K,           FLASH_DATA,  YES,   YES ); // Secret + Boardloader  !!write <-no 
+  SET_REGION( 1, FLASH_BASE_S + SIZE_64K,  SIZE_128K,          FLASH_CODE,  YES,   YES ); // Bootloader code  !!write <- no
```
[the remark this answers is not in the document]
Comment (cepetr) [reply core/embed/trezorhal/stm32u5/mpu.c]: fixed, see 853b1c34b1de7598c26b3671bbade6b27b0984af

File: core/embed/trezorhal/stm32u5/mpu.c [hunk 3]
```diff
@@ -18,241 +18,163 @@
...
+  mpu_set_attributes();
+  // clang-format off
+  //   REGION    ADDRESS                   SIZE                TYPE       WRITE   UNPRIV 
+  SET_REGION( 0, FLASH_BASE_S + SIZE_192K, SIZE_128K,          FLASH_DATA,  YES,   YES ); // Storage 
+  SET_REGION( 1, FLASH_BASE_S + SIZE_320K, SIZE_3776K,         FLASH_CODE,   NO,   YES ); // Firmware 
+  SET_REGION( 2, SRAM1_BASE_S,             SIZE_768K,          SRAM,        YES,   YES ); // SRAM1 
+  SET_REGION( 3, SRAM2_BASE_S + 0x100,     SIZE_1728K - 0x100, SRAM,        YES,   YES ); // SRAM2/3/5 + stack guard
+  SET_REGION( 4, GFXMMU_BUFFERS_S,         SIZE_16M,           SRAM,        YES,   YES ); // Frame buffer 
+  SET_REGION( 5, PERIPH_BASE_S,            SIZE_256M,          PERIPHERAL,  YES,   YES ); // Peripherals 
+  SET_REGION( 6, FLASH_OTP_BASE,           FLASH_OTP_SIZE,     FLASH_DATA,  YES,   YES ); // OTP DIS_REGION( 7 );
```
[the remark this answers is not in the document]
Comment (cepetr) [reply core/embed/trezorhal/stm32u5/mpu.c]: fixed, see 853b1c34b1de7598c26b3671bbade6b27b0984af

File: core/embed/trezorhal/stm32u5/mpu.c [hunk 4]
```diff
@@ -18,241 +18,163 @@
...
-  MPU->RBAR = FLASH_OTP_BASE | LL_MPU_REGION_ALL_RW |
-              LL_MPU_INSTRUCTION_ACCESS_DISABLE | SHAREABILITY_FLASH;
-  MPU->RLAR = REGION_END(FLASH_OTP_BASE + FLASH_OTP_SIZE - 1) |
-              ATTR_IDX_FLASH_NON_CACHABLE;
-
-  // Enable MPU
+  mpu_set_attributes();
+  // clang-format off
+  //   REGION    ADDRESS                   SIZE                TYPE       WRITE   UNPRIV 
+  SET_REGION( 0, FLASH_BASE_S,             SIZE_64K,           FLASH_DATA,  YES,   YES ); // Secret + Boardloader  !!write <-no
```
[the remark this answers is not in the document]
Comment (cepetr) [reply core/embed/trezorhal/stm32u5/mpu.c]: ok, I understand, so no change is needed.

File: core/embed/trezorhal/stm32u5/mpu.c [hunk 5]
```diff
@@ -18,241 +18,163 @@
...
+  mpu_set_attributes();
+  // clang-format off
+  //   REGION    ADDRESS                   SIZE                TYPE       WRITE   UNPRIV 
+  SET_REGION( 0, FLASH_BASE_S,             SIZE_16K,           FLASH_DATA,  YES,   YES ); // Secret 
+  SET_REGION( 1, FLASH_BASE_S + SIZE_16K,  SIZE_48K,           FLASH_CODE,   NO,   YES ); // Boardloader code 
+  SET_REGION( 2, FLASH_BASE_S + SIZE_64K,  SIZE_4032K,         FLASH_DATA,  YES,   YES ); // Bootloader + Storage + Firmware 
+  SET_REGION( 3, SRAM1_BASE_S,             SIZE_768K,          SRAM,        YES,   YES ); // SRAM1 
+  SET_REGION( 4, SRAM2_BASE_S + 0x100,     SIZE_1728K - 0x100, SRAM,        YES,   YES ); // SRAM2/3/5 + stack guard 
+  SET_REGION( 5, GFXMMU_BUFFERS_S,         SIZE_16M,           SRAM,        YES,   YES ); // Frame buffer 
+  SET_REGION( 6, PERIPH_BASE_NS,           SIZE_512M,          PERIPHERAL,  YES,   YES ); // Peripherals NS + S ?? proc NS
```
[the remark this answers is not in the document]
Comment (cepetr) [reply core/embed/trezorhal/stm32u5/mpu.c]: ok, I shrunk it to 256MB, see 853b1c34b1de7598c26b3671bbade6b27b0984af

**Review by TychoVrahe — APPROVED** [review]
lgtm, please fix the residual style issues, squash and add [no changelog] to the commit message

**Pull request merged**